core/src/test/java/org/apache/iceberg/io/TestResolvingIO.java

```diff
@@ -47,4 +60,35 @@ public void testResolvingFileIOJavaSerialization() throws IOException, ClassNotF
     FileIO roundTripSerializedFileIO = TestHelpers.roundTripSerialize(testResolvingFileIO);
     assertThat(roundTripSerializedFileIO.properties()).isEqualTo(testResolvingFileIO.properties());
   }
+
+  @Test
+  public void resolveFileIOBulkDeletion() throws IOException {
+    // configure resolving fileIO
+    FileIO resolvingFileIO = new ResolvingFileIO();
```

Comment:
```suggestion
    ResolvingFileIO resolvingFileIO = new ResolvingFileIO();
```
then you don't need to perform the 2 casts to `HadoopConfigurable` and `SupportsBulkOperations` further below